### PR TITLE
Initial Pipeline

### PR DESCRIPTION
**Pull request by rebeccafair**
See https://github.com/pace-neutrons/Pace-Project-Plan/issues/120 for requirements

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+}
+
+properties([
+  parameters([
+    string(
+      defaultValue: get_matlab_version(env.JOB_BASE_NAME),
+      description: 'The version of Matlab to use e.g. 2019b.',
+      name: 'MATLAB_VERSION',
+      trim: true
+    ),
```
Comment (hsautessella) [inline Jenkinsfile:68]: We've recently started declaring our parameter defaults differently, due to some issues when changing them. Have a look at this issue https://github.com/pace-neutrons/horace/issues/574. The issue boils down to the defaults only being updated _after_ the pipeline starts.

These are the PRs for Herbert and Horace: https://github.com/pace-neutrons/Herbert/pull/307 https://github.com/pace-neutrons/Horace/pull/587 resolving said issue.

File: powershell_scripts/execute_matlab_tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,27 @@
+. $PSScriptRoot/powershell_helpers.ps1 <# Imports:
```
Comment (hsautessella) [inline powershell_scripts/execute_matlab_tests.ps1:1]: This script has a number of environment variables it relies on. Either making the variables parameters, or documenting what variables are required, would make running (and debugging) this locally easier

**Review by hsautessella — COMMENTED** [review]
Just a couple of comments on Jenkins and PowerShell parameters, but I think this looks good

**Review by mducle — APPROVED** [review]
All looks good... just a small nitpicky comment.

File: EuphonicDisp2SqwTest.m [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+            expected_wsim = read_sqw('quartz/expected_cut1d_disp2sqw_tobyfit.sqw');
+
+            % Tobyfit results are non-deterministic and have high errors
+            % when running such a quick test case, allow generous relative
+            % errors
+            disp('Testing result...');
+            import matlab.unittest.constraints.IsEqualTo
+            import matlab.unittest.constraints.AbsoluteTolerance
+            import matlab.unittest.constraints.RelativeTolerance
+            bounds = AbsoluteTolerance(1e-10*mean(expected_wsim.data.s)) | RelativeTolerance(1e-5);
```
[the remark this answers is not in the document]
Comment (rebeccafair) [reply EuphonicDisp2SqwTest.m:81]: Good spot - I forgot to remove that comment. And indentation fixed.

**Pull request merged**